anyblok/bloks/anyblok_core/system/sequence.py

```diff
@@ -161,10 +161,10 @@ def nextval(self):
         """
         if self.no_gap:
             self.refresh(with_for_update={"nowait": True})
-            nextval = self.number + 1
+            nextval = self.current + 1 if self.current else self.start
```

Comment:
I would prefer to test if self.current `is not None` but it's quite a details:

```suggestion
            nextval = self.current + 1 if self.current is not None else self.start
```